app/src/features/surveys/view/survey-animals/form-sections/MeasurementAnimalForm.tsx

```diff
@@ -3,124 +3,63 @@ import CbSelectField from 'components/fields/CbSelectField';
-              <MeasurementFormContent key={`${measurement._id}`} index={index} measurements={measurements} />
-            ))}
-          </FormSectionWrapper>
-        </>
-      )}
-    </FieldArray>
-  );
-};
-
 interface MeasurementFormContentProps {
   index: number;
   measurements?: IMeasurementStub[];
+  mode: ANIMAL_FORM_MODE;
 }
 
-const MeasurementFormContent = ({ index, measurements }: MeasurementFormContentProps) => {
+export const MeasurementAnimalFormContent = (props: MeasurementFormContentProps) => {
+  const { index, measurements, mode } = props;
+  const name: keyof IAnimal = 'measurements';
   const { values, handleChange, setFieldValue, handleBlur } = useFormikContext<IAnimal>();
-  const [measurement, setMeasurement] = useState<IMeasurementStub>();
+  const taxonMeasurementId = values.measurements?.[index]?.taxon_measurement_id;
+  const [currentMeasurement, setCurrentMeasurement] = useState<IMeasurementStub | undefined>(
+    measurements?.find((lookup_measurement) => lookup_measurement.taxon_measurement_id === taxonMeasurementId)
+  );
+  const isQuantMeasurement = has(currentMeasurement, 'unit');
 
-  const taxonMeasurementId = values.measurements[index].taxon_measurement_id;
-  const isQuantMeasurement = has(measurement, 'unit');
+  const taxonMeasurementIDName = getAnimalFieldName<IAnimalMeasurement>(name, 'taxon_measurement_id', index);
+  const valueName = getAnimalFieldName<IAnimalMeasurement>(name, 'value', index);
+  const optionName = getAnimalFieldName<IAnimalMeasurement>(name, 'qualitative_option_id', index);
 
-  const tMeasurementIDName = getAnimalFieldName<IAnimalMeasurement>(NAME, 'taxon_measurement_id', index);
-  const valueName = getAnimalFieldName<IAnimalMeasurement>(NAME, 'value', index);
-  const optionName = getAnimalFieldName<IAnimalMeasurement>(NAME, 'qualitative_option_id', index);
+  useEffect(() => {
+    setCurrentMeasurement(measurements?.find((m) => m.taxon_measurement_id === taxonMeasurementId));
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [measurements]); //Sometimes will not display the correct fields without this useEffect but could have side effects, may need to revisit.
 
   const handleMeasurementTypeChange = (event: SelectChangeEvent<unknown>) => {
     handleChange(event);
     setFieldValue(valueName, '');
     setFieldValue(optionName, '');
     const m = measurements?.find((m) => m.taxon_measurement_id === event.target.value);
-    setMeasurement(m);
+    setCurrentMeasurement(m);
+    handleMeasurementName('', m?.measurement_name ?? '');
   };
 
   const validateValue = async (val: '' | number) => {
-    const min = measurement?.min_value ?? 0;
-    const max = measurement?.max_value;
-    const unit = measurement?.unit ? ` ${measurement.unit}'s` : ``;
+    const min = currentMeasurement?.min_value ?? 0;
+    const max = currentMeasurement?.max_value;
+    const unit = currentMeasurement?.unit ? ` ${currentMeasurement.unit}'s` : ``;
```

Comment:
I don't think units need to be plural. "Measurement must be less than 12 cm" is better than "Measurement must be less than 12 cm's" I think.

Reply:
Yea I think this makes sense. Updated